                                | Column            | Scale of Measurement|
                                |---------------    | --------------------|
                                | enrollment_id     | Nominal             |
                                | course_name       | Nominal             |
                                | enrollment_year   | Interval            |
                                | final_grade       | Ordinal             |
                                | hours_studied     | Ratio               |
                                | completion_status | Nominal             |



In [1]:
import pandas as pd
import numpy as np


In [31]:
skyline = pd.read_csv("skyline_enrollments.csv")

In [32]:
skyline["Course_name"]

0               SQL Basics
1           Statistics 101
2               SQL Basics
3     Python for Beginners
4     Tableau Fundamentals
              ...         
95          Statistics 101
96    Tableau Fundamentals
97    Python for Beginners
98      Intro to Analytics
99      Intro to Analytics
Name: Course_name, Length: 100, dtype: str

In [33]:
len(skyline["Course_name"])

100

In [34]:
Course_name = skyline["Course_name"] .mode() ; print( f"Mode Course_name: {Course_name.values}")

Mode Course_name: <StringArray>
['Tableau Fundamentals']
Length: 1, dtype: str


In [35]:
skyline["Completion_status"]

0     In Progress
1     In Progress
2       Completed
3     In Progress
4     In Progress
         ...     
95      Completed
96    In Progress
97      Completed
98      Completed
99    In Progress
Name: Completion_status, Length: 100, dtype: str

In [36]:
len(skyline["Completion_status"])

100

In [37]:
Completion_status = skyline["Completion_status"] .mode() ; print( f"Mode Completion_status: {Completion_status.values}")

Mode Completion_status: <StringArray>
['Completed']
Length: 1, dtype: str


In [41]:
skyline["Completion_status"].nunique()

3

In [44]:
skyline["Enrollment_ids"].nunique()

100

In [45]:
len(skyline["Enrollment_ids"])

100

In [48]:
Enrollment_ids = skyline["Enrollment_ids"] .mode() ; print( f"Mode Enrollment_ids: {Enrollment_ids.values}")

Mode Enrollment_ids: <StringArray>
['ENR-1027', 'ENR-1108', 'ENR-1166', 'ENR-1205', 'ENR-1210', 'ENR-1267',
 'ENR-1344', 'ENR-1347', 'ENR-1521', 'ENR-1602', 'ENR-1625', 'ENR-1719',
 'ENR-1723', 'ENR-1850', 'ENR-1857', 'ENR-1978', 'ENR-2144', 'ENR-2200',
 'ENR-2353', 'ENR-2428', 'ENR-2435', 'ENR-2444', 'ENR-2519', 'ENR-2671',
 'ENR-2764', 'ENR-3068', 'ENR-3207', 'ENR-3212', 'ENR-3329', 'ENR-3519',
 'ENR-3615', 'ENR-3683', 'ENR-3787', 'ENR-3909', 'ENR-3973', 'ENR-4097',
 'ENR-4165', 'ENR-4389', 'ENR-4754', 'ENR-4823', 'ENR-5016', 'ENR-5200',
 'ENR-5290', 'ENR-5347', 'ENR-5367', 'ENR-5385', 'ENR-5455', 'ENR-5713',
 'ENR-5780', 'ENR-5998', 'ENR-6165', 'ENR-6168', 'ENR-6442', 'ENR-6468',
 'ENR-6487', 'ENR-6506', 'ENR-6520', 'ENR-6524', 'ENR-6583', 'ENR-6627',
 'ENR-6765', 'ENR-6767', 'ENR-6955', 'ENR-7136', 'ENR-7186', 'ENR-7202',
 'ENR-7211', 'ENR-7351', 'ENR-7409', 'ENR-7524', 'ENR-7647', 'ENR-7672',
 'ENR-7873', 'ENR-8055', 'ENR-8115', 'ENR-8116', 'ENR-8147', 'ENR-8282',
 'ENR-8345', 'EN

In [50]:
skyline["Final_grade"].value_counts()

Final_grade
D    24
B    23
F    20
A    17
C    16
Name: count, dtype: int64

In [51]:
len(skyline["Final_grade"])

100

In [52]:
Final_grade = skyline["Final_grade"] .mode() ; print( f"Mode Final_grade: {Final_grade.values}")

Mode Final_grade: <StringArray>
['D']
Length: 1, dtype: str


In [80]:
grade_mapping = {"A": 1, "B": 2, "C": 3, "D": 4, "F": 5}



In [81]:
final_grade_map = skyline["Final_grade"].map(grade_mapping)


In [79]:
print(final_grade_map)
print(grade_mapping)

0     4
1     1
2     2
3     2
4     1
     ..
95    4
96    4
97    3
98    2
99    2
Name: Final_grade, Length: 100, dtype: int64
{'A': 1, 'B': 2, 'C': 3, 'D': 4, 'F': 5}


In [82]:
grade_median = final_grade_map.median()
print(f"Median Final_grade: {grade_median}")

Median Final_grade: 3.0


In [101]:
Final_grade_mean = final_grade_map.mean()
print(f"Mean Final_grade: {Final_grade_mean:.2f}")

Mean Final_grade: 3.07


## Final_grade

The numbers assigned to each grade (A=1 through F=5) were only used to establish rank order for mapping purposes — they don't represent equal, real differences in student performance. Because of this, the mean of 3.07 is mathematically correct but doesn't have a clear, trustworthy real-world meaning; reporting it to a stakeholder would require noting this limitation.


In [85]:
enrollment_year_mean = skyline["enrollment_year"].mean()
print(f"Mean enrollment year: {enrollment_year_mean:.2f}")

Mean enrollment year: 2023.82


In [88]:
enrollment_year_median = skyline["enrollment_year"].median()
print(f"Median enrollment year: {enrollment_year_median:.2f}")

Median enrollment year: 2024.00


In [89]:
enrollment_year_mode = skyline["enrollment_year"].mode()[0]
print(f"Mode enrollment year: {enrollment_year_mode}")

Mode enrollment year: 2023


In [90]:
enrollment_year_std = skyline["enrollment_year"].std()
print(f"Standard deviation of enrollment years: {enrollment_year_std:.2f}")


Standard deviation of enrollment years: 1.35


In [92]:
enrollment_year_q1 = skyline["enrollment_year"].quantile(0.25)
enrollment_year_q3 = skyline["enrollment_year"].quantile(0.75)
enrollment_year_iqr = enrollment_year_q3 - enrollment_year_q1

print(f"Q1 (25th percentile): {enrollment_year_q1:.2f}")
print(f"Q3 (75th percentile): {enrollment_year_q3:.2f}")
print(f"IQR: {enrollment_year_iqr:.2f}")

Q1 (25th percentile): 2023.00
Q3 (75th percentile): 2025.00
IQR: 2.00


In [96]:
Hours_studied_mean = skyline["Hours_studied"].mean()
print(f"Mean Hours studied: {Hours_studied_mean:.2f}")

Mean Hours studied: 30.18


In [97]:
Hours_studied_median = skyline["Hours_studied"].median()
print(f"Median Hours studied: {Hours_studied_median:.2f}")

Median Hours studied: 30.06


In [98]:
Hours_studied_mode = skyline["Hours_studied"].mode()[0]
print(f"Mode Hours studied: {Hours_studied_mode}")

Mode Hours studied: 27.77


In [99]:
Hours_studied_std = skyline["Hours_studied"].std()
print(f"Standard deviation of Hours studied: {Hours_studied_std:.2f}")


Standard deviation of Hours studied: 5.20


In [100]:
Hours_studied_q1 = skyline["Hours_studied"].quantile(0.25)
Hours_studied_q3 = skyline["Hours_studied"].quantile(0.75)
Hours_studied_iqr = Hours_studied_q3 - Hours_studied_q1

print(f"Q1 (25th percentile): {Hours_studied_q1:.2f}")
print(f"Q3 (75th percentile): {Hours_studied_q3:.2f}")
print(f"IQR: {Hours_studied_iqr:.2f}")

Q1 (25th percentile): 26.99
Q3 (75th percentile): 34.25
IQR: 7.26


## Hours_studied

The mean (30.18) and median (30.06) are very close, suggesting the distribution is symmetrical around the center, with no significant skew. This makes sense since Hours_studied was generated from a normal distribution.